src/details/ArborX_DetailsCrsGraphWrapperImpl.hpp

```diff
@@ -59,6 +59,9 @@ struct InsertGenerator
   CountView _counts;
   PermutedOffset _permuted_offset;
 
+  using TreeTraversalControl =
+      detected_t<CallbackTreeTraversalControlArchetypeAlias, Callback>;
+
```

Comment:
OK, seeing this makes me question the semantics of this kind of traversal when storing the results. What does it mean for a user to ask for a quick/pair traversal? Specifically,
- Is it deterministic?
  If it's not, the result of the second rerun may differ from the first.
- Is it balanced?
  If it's not, should a user have no expectation at all?
In general, this implicitly ties the result to the implementation details of such a traversal. If you change the details to, for example, `query_index >= node_index`, this would completely change the result a user gets. Is this something we want?

Reply:
Agreed. Conceptually, I'm totally fine with the pure callback, but these questions came to mind when thinking about crs.